include/library/spdm_requester_lib.h

```diff
@@ -525,6 +525,33 @@ libspdm_return_t libspdm_start_session(void *spdm_context, bool use_psk,
                                        uint8_t *heartbeat_period,
                                        void *measurement_hash);
 
+/**
+ * This function retrieves the supported algorithms from the responder.
+ * It sends the GET_VERSION and GET_CAPABILITIES requests, where GET_CAPABILITIES.Param1[0] is set.
+ * If the Responder supports this extended capability, the Responder will include the Supported
+ * Algorithms Block in its CAPABILITIES response.
+ *
+ * @param spdm_context                          A pointer to the SPDM context.
+ * @param responder_supported_algorithms_length  On input, indicates the size in bytes of the provided buffer.
+ *                                              The buffer must be at least sizeof(libspdm_responder_supported_algorithms_13_t) bytes.
+ *                                              On output, the size in bytes of the supported algorithms data.
+ * @param responder_supported_algorithms_buffer  A pointer to a destination buffer to store the supported algorithms.
+ *                                              Must not be NULL. The buffer must be large enough to hold the supported algorithms data.
+ * @param spdm_version                          A pointer to store the SPDM version used for the request.
+ *
+ * @retval RETURN_SUCCESS                        The supported algorithms were successfully retrieved.
+ * @retval RETURN_DEVICE_ERROR                   A device error occurs when communicates with the device.
+ * @retval RETURN_UNSUPPORTED                    The operation is not supported by the device.
+ * @retval RETURN_SECURITY_VIOLATION             Any verification fails.
+ *
+ * @note   The buffer must be at least sizeof(libspdm_responder_supported_algorithms_13_t) bytes.
+ *         The function will assert if responder_supported_algorithms_buffer is NULL.
+ */
+libspdm_return_t libspdm_get_supported_algorithms(void *spdm_context,
+                                                  size_t *responder_supported_algorithms_length,
+                                                  void *responder_supported_algorithms_buffer,
+                                                  uint8_t *spdm_version);
```

Comment:
If you include `spdm_version` to `libspdm_responder_supported_algorithms_13_t`, then you can remove `spdm_version` parameter here.

Reply:
>then you can remove spdm_version parameter here.

The Integrator still needs to know what `libspdm_responder_supported_algorithms_*_t` to apply to `responder_supported_algorithms_buffer`. Without this parameter they would need to `libspdm_get_data` `LIBSPDM_DATA_SPDM_VERSION`, which is not the end of the world, but this makes it more convenient.